### PR TITLE
Added debugging tools for POST based data senders

### LINKED ISSUE #1064: EDDN: CL arg and simple listener to properly debug messages
1. CL arg to change the destination of EDDN messages to, e.g., http://localhost:9000/
2. A listener on that address that will just log all the messages to a file

Then we can properly test/debug EDDN sending without running into issues like someone else happening to dock at the same station, or the gateway's duplication detection dropping the message.

### PR DESCRIPTION
**Pull request by A-UNDERSCORE-D**
A simple HTTP handler has been added that will dump any and all POST
data it gets to EDMC's log.

Additionally, a new command line argument to switch EDMC to using this
as its EDDN target has been added.

!! This needs testing before merging.

Closes #1064

### CODE REVIEW

**Comment by Athanasius** [comment]
Ooooh, no, I didn't want this going into application logging, it needs to be the raw data out to its own file.

`%TEMP/EDMarketConnector-EDDN-log.txt` or the like.

**Comment by Athanasius** [comment]
> Ooooh, no, I didn't want this going into application logging, it needs to be the raw data out to its own file.
> 
> `%TEMP/EDMarketConnector-EDDN-log.txt` or the like.

To be clear, this goes into its own file so it's pure JSONL, without the usual logging timestamp/module prefixing.

**Comment by A-UNDERSCORE-D** [comment]
oh I can do that too. One moment

**Comment by Athanasius** [comment]
This should be extended for a CL arg per other thing we send to.

EDSM
Inara

*Maybe* also the 'providers' ?

The basic idea being able to test functionality without sending possibly bad data to the live services.  Make debugging easier.

**Comment by A-UNDERSCORE-D** [comment]
I've replaced the original implementation with support for arbitrary POSTs, on disk they're organized by path (please check anti-traversal magic, it should be fine.) and configured via CL arg that is repeated for any name you want to support.

**Comment by A-UNDERSCORE-D** [comment]
Rebased against develop

File: PLUGINS.md [hunk 1]
```diff
@@ -1043,6 +1043,39 @@ step for the extra module(s) until it works.
...
+
+To add support for said debug webserver to your plugin, you need to check `config.debug_senders` (`list[str]`) for
+some indicator string for your plugin. `debug_senders` is generated from args to `--debug-sender` on the invocation
+command line.
+
+If said string exists, `DEBUG_WEBSERVER_HOST` and `DEBUG_WEBSERVER_PORT` in
+`edmc_data` will contain the host and port for the currently running local webserver. Simply redirect your requests
+there, and your requests will be logged to disk. For organisation, rewrite your request path to simply be `/pluginname`.
+
+logs exist in `$TEMP/EDMarketConnector/http_debug/$path.log`. If somehow you manage to cause a directory traversal, your
```
Comment (Athanasius) [inline PLUGINS.md:1058]: s/logs/Logs/

File: debug_webserver.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            try:
+                edsm_data = extract_edsm_data(data)
+                data = data + "\n" + json.dumps(edsm_data)
+            except Exception:
+                pass
+
+        target_file = output_data_path / (safe_file_name(target_path) + '.log')
+        if target_file.parent != output_data_path:
+            logger.warning(f"REFUSING TO WRITE FILE THAT ISN'T IN THE RIGHT PLACE! {target_file=}")
+            logger.warn(f'DATA FOLLOWS\n{data}')
```
Comment (Athanasius) [inline debug_webserver.py:66]: Shoud be logger.warning

File: debug_webserver.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            target_path = 'WEB_ROOT'
+
+        response: Union[Callable[[str], str], str, None] = DEFAULT_RESPONSES.get(target_path)
+        if callable(response):
+            response = response(data)
+
+        self.send_response_only(200, "OK")
+        if response is not None:
+            self.send_header('Content-Length', str(len(response)))
+        self.end_headers()  # This is needed because send_response_only DOESN'T ACTUALLY SEND THE RESPONSE </rant>
```
Comment (Athanasius) [inline debug_webserver.py:50]: Blank line after conditional

File: plugins/edsm.py [hunk 4]
```diff
@@ -528,7 +529,13 @@ def cmdr_data(data: CAPIData, is_beta: bool) -> None:
...
 
 
+TARGET_URL = 'https://www.edsm.net/api-journal-v1'
+if 'edsm' in debug_senders:
+    TARGET_URL = f'http://{DEBUG_WEBSERVER_HOST}:{DEBUG_WEBSERVER_PORT}/edsm'
+
 # Worker thread
+
+
 def worker() -> None:
```
Comment (Athanasius) [inline plugins/edsm.py:539]: Might as well # noqa this.

At some point in the future we'll get around to grepping for all such and see about refactors.  Reduce the noise for now.

**Review by Athanasius — CHANGES_REQUESTED** [review]
Only cosmetics for now.  I should actually try out the code ....

**Comment by Athanasius** [comment]
And now I've actually run it, including debugging both EDDN and Inara at the same time, yup, looks good once those niggles are addressed.

**Review by Athanasius — APPROVED** [review]

**Pull request merged**